### ISSUE TITLE
Temperature Sensor Details

### ISSUE THREAD

**Issue by nbars** [techdregs/Breville_Bambino_Data#1]
Hey, thanks for documenting the whole process of disassembling the Bambino! I am currently building custom firmware (based on your tear down) for the Bambino, and I would like to know if you have any further details regarding the temperature sensor and how it's working.
<img width="564" alt="Screenshot 2024-02-04 at 18 20 44" src="https://github.com/techdregs/Breville_Bambino_Data/assets/1810786/70eddaae-9d78-4524-a316-5aee3d65c0fc">

The NTC_Sig (PB2) port is not connected to the ADC; Hence, I assume that PIN19 (PB1) is used to measure the sensor's resistance. Unfortunately, I am inexperienced with hardware-related things, and I'd like to know whether there are more details on calculating the actual temperature based on the current ADC value and whether I still need to consider NTC_Sig during my measurement. Based on the datasheet of other thermistors, I would have expected a simple voltage divider here, but for me, this looks a little more complicated.

Best,
Nils

**Comment by techdregs** [comment]
Nils,

Well, the simplest explanation I can think of is that I may have the pins mislabeled. I would check that on your board; perhaps I got them reversed? They are right next to each other on the MCU, after all.

With a NTC, that would make the most sense by far, and that's why I labeled that NTC_sig. I didn't look closely at the datasheet to know that pin20 wasn't on the ADC, so I bet I just got them switched on the schematic. 

Please let me know if that isn't the case. I can't check, as I sold this machine to a friend a couple of months ago.

**Comment by nbars** [comment]
Hey, thanks for the response. A quick check indicates that none of the (microcontroller) pins are directly connected to the NTC (opposing the schematics). How did you do all the probing? Are you using some special tools? I have a really hard time identifying the pins I am currently probing. Maybe I just need a better magnifier :) 

Also, do you have any clue why there are two GPIOs involved?

I will report back here as soon as I have time to do the tracing properly.

**Comment by techdregs** [comment]
> Hey, thanks for the response. A quick check indicates that none of the (microcontroller) pins are directly connected to the NTC (opposing the schematics). How did you do all the probing? Are you using some special tools? I have a really hard time identifying the pins I am currently probing. Maybe I just need a better magnifier :)
> 
> Also, do you have any clue why there are two GPIOs involved?
> 
> I will report back here as soon as I have time to do the tracing properly.

No, I really am not sure why they have two. The first would obviously need to be sensing the voltage divider, but the other might simply be a check for signal or calibration somehow. I'm not experienced enough to understand the need for the second pin.

As for tools, I just used a regular multimeter with some sharp probes. There some conformal coating on stuff that you have to make sure you poke through to get a good reading. It does take a bit to double and triple check stuff... and I would be shocked if my schematic is 100% accurate, given the complexity of the circuit and difficulties probing those little pins.

**Comment by nbars** [comment]
Hey, the pin labeled NTC_Sig is connected to the right side of R106. Also, I am now using PIN19 as a signal, which works fine. Furthermore, the main connector to the display unit is labeled in reverse order, i.e., the first pin is the last one, and so on.

Thanks for the nice schematics! They have been really helpful, and I have already managed to control each component.

**Comment by techdregs** [comment]
> Hey, the pin labeled NTC_Sig is connected to the right side of R106. Also, I am now using PIN19 as a signal, which works fine. Furthermore, the main connector to the display unit is labeled in reverse order, i.e., the first pin is the last one, and so on.
> 
> Thanks for the nice schematics! They have been really helpful, and I have already managed to control each component.

Interesting. Is Pin 19 then going above R105 instead of below it? I would think it would have to be in order to take a correct signal. I would guess NTC_Sig would then just be serving to check if there's an NTC connected and seems to have some reverse polarity protection or something.

For the display, do you mean CON2 or CON1? It's been a while, but IIRC, for that one it was just a matter of orientation (how you held it in your hand). I thought I labeled the board side as if that connector was on the bottom right, but the front PCB I don't recall how I was holding it. It's certainly possible I flipped those.

**Comment by nbars** [comment]
As far as I can remember, PIN19 is connected, just like you depicted. Right now, I am in the process of programming the machine, and for safety, I assembled it again, so I cannot trace the layout in more detail.

Regarding the connector, I am talking about CON2. It has pin one marked with a white circle, considering that the pin numbering order must be reversed on the schematic. If you only consider the physical orientation, your figure is correct, but I don't think this makes sense, do you?

**Comment by inma30** [comment]
Hi both, I wish i'd seen this thread when I was looking at the circuit last month.
I design electronics for a living, albeit for the automotive sector.
The circuit is fairly standard, it is a resistor divider, with the thermistor and R78 forming the divider.
R105 and C43 are a low pass filter into the ADC (I assumed it, looks like you confirmed it).
NTC_Sig is a puzzle, I didn't get as far as looking at the micro pins and expected it to be another analogue input. The fact that there is no resistor inline suggests they are using it as an output, perhaps to test the ADC is working, maybe also to confirm an open circuit thermistor fault.
D72 and D73 are a diode clamp, quite hefty diodes too, so I expect R106 is very low resistance, possibly 0R.
My interest was a mod to increase the brew temperature.

**Comment by techdregs** [comment]
> Hi both, I wish i'd seen this thread when I was looking at the circuit last month. I design electronics for a living, albeit for the automotive sector. The circuit is fairly standard, it is a resistor divider, with the thermistor and R78 forming the divider. R105 and C43 are a low pass filter into the ADC (I assumed it, looks like you confirmed it). NTC_Sig is a puzzle, I didn't get as far as looking at the micro pins and expected it to be another analogue input. The fact that there is no resistor inline suggests they are using it as an output, perhaps to test the ADC is working, maybe also to confirm an open circuit thermistor fault. D72 and D73 are a diode clamp, quite hefty diodes too, so I expect R106 is very low resistance, possibly 0R. My interest was a mod to increase the brew temperature.

Good inputs! I'm just a hobbyist, so there are many times when I'm just not exactly sure what some of this stuff is for. And, as I mentioned above, since I traced all this stuff out by eye and a multimeter, I'm sure there are some errors in the schematic.

**Comment by inma30** [comment]
It made my day to find this schematic, thank you.

**Comment by banoz** [comment]
Lots of good info can be picked from BES880 service manual. PB2 signal might be intended to be used as a calibration point.

**Comment by banoz** [comment]
Based on BB+ measurements R78 is 22K, R106 is 2.2K, R105 is 1K. When PB2 is pulled down R78 and R106 form 2K resistor in parallel. This provides 127 mV on 25C, given the NTC is 50kΩ @25°C, ß25/80=3976K. The NTC params are very similar to NRNE504F3975B2F, and using datapoints from datasheet to model the logarithmic curve we can get a good fit on 80-100C by using **a = -29.66**, **b = 136.62**.

**Comment by banoz** [comment]
<img width="564" alt="bb_ntc" src="https://github.com/user-attachments/assets/b4bf7515-fea5-4226-a6be-882445d67872">
